src/ol_infrastructure/applications/mitxonline/__main__.py

```diff
@@ -0,0 +1,164 @@
+"""Create the infrastructure and services needed to support the MITx Online application.
+
+- Create a Redis instance in AWS Elasticache
+- Create a PostgreSQL database in AWS RDS for production environments
+- Create an IAM policy to grant access to S3 and other resources
+"""
+import json
+
+import pulumi_vault as vault
+from pulumi import Config, StackReference, export
+from pulumi_aws import ec2, iam, s3
+
+from ol_infrastructure.components.aws.database import OLAmazonDB, OLPostgresDBConfig
+from ol_infrastructure.components.services.vault import (
+    OLVaultDatabaseBackend,
+    OLVaultPostgresDatabaseConfig,
+)
+from ol_infrastructure.lib.aws.iam_helper import lint_iam_policy
+from ol_infrastructure.lib.ol_types import AWSBase
+from ol_infrastructure.lib.pulumi_helper import parse_stack
+from ol_infrastructure.lib.stack_defaults import defaults
+
+mitxonline_config = Config("mitxonline")
+stack_info = parse_stack()
+network_stack = StackReference(f"infrastructure.aws.network.{stack_info.name}")
+apps_vpc = network_stack.require_output("applications_vpc")
+operations_vpc = network_stack.require_output("operations_vpc")
+aws_config = AWSBase(
+    tags={
+        "OU": "mitxonline",
+        "Environment": f"applications_{stack_info.env_suffix}",
+        "Application": "mitxonline",
+    }
+)
+
+# Create S3 bucket
+
+# Bucket used to store files from MITx Online app.
+mitxonline_bucket_name = f"ol-mitxonline-app-{stack_info.env_suffix}"
+mitxonline_bucket = s3.Bucket(
+    f"mitxonline-{stack_info.env_suffix}",
+    bucket=mitxonline_bucket_name,
+    versioning=s3.BucketVersioningArgs(
+        enabled=True,
+    ),
+    tags=aws_config.tags,
+)
+
+
+mitxonline_iam_policy = iam.Policy(
+    f"mitxonline-{stack_info.env_suffix}-policy",
+    description="AWS access controls for the MITx Online application in the "
+    f"{stack_info.name} environment",
+    path=f"/ol-applications/mitxonline/{stack_info.env_suffix}/",
+    name_prefix="aws-permissions-",
```

Comment:
The name prefix should be more descriptive, as this is what shows up in the console for being able to identify what the policy is. Something like `mitxonline-{stack_info.env_suffix}-application-permissions-` or `mitxonline-{stack_info.env_suffix}-application-policy-`